components/sx127x.rst

```diff
@@ -0,0 +1,380 @@
+
+``sx127x.send_packet`` **Action**
+
+This :ref:`action <config-action>` sends a packet, the ``sx127x`` needs to be in packet mode.
+
+.. code-block:: yaml
+
+    on_...:
+      - sx127x.send_packet:
+          data: [0x1F, 0x3E, 0x06, 0x5F, 0x4F, 0x5F, 0xAC, 0xB1]
+
+Configuration variables:
+
+- **data** (**Required**, list): The packet to send, length should match the configured
+  payload_length.
+
+LoRa:
+-----
+
+LoRa example using an explicit header, spreading factor 7 and coding rate 4/5.
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    sx127x:
+      dio0_pin: GPIO26
+      cs_pin: GPIO18
+      rst_pin: GPIO23
+      pa_pin: BOOST
+      pa_power: 4
+      bandwidth: 125_0kHz
+      crc_enable: true
+      frequency: 433920000
+      modulation: LORA
+      rx_start: true
+      sync_value: 0x33
+      spreading_factor: 7
+      coding_rate: CR_4_5
+      on_packet:
+        then:
+          - lambda: |-
+              ESP_LOGD("lambda", "packet %s", format_hex(x).c_str());
+              ESP_LOGD("lambda", "rssi %.2f", rssi);
+              ESP_LOGD("lambda", "snr %.2f", snr);
+
+      button:
+        - platform: template
+          name: "Transmit Packet"
+          on_press:
+            then:
+              - sx127x.send_packet:
+                 data: [0xC5, 0x51, 0x78, 0x82, 0xB7, 0xF9, 0x9C, 0x5C]
+
+
+FSK/OOK Packet Mode:
+--------------------
+
+FSK packet mode example using a bit rate of 4800.
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    sx127x:
+      cs_pin: GPIO18
+      rst_pin: GPIO23
+      dio0_pin: GPIO26
+      pa_pin: BOOST
+      pa_power: 17
+      bitrate: 4800
+      bitsync: true
+      crc_enable: true
+      frequency: 433920000
+      modulation: FSK
+      rx_start: true
+      payload_length: 8
+      sync_value: [0x33, 0x33]
+      preamble_size: 2
+      preamble_errors: 8
+      preamble_polarity: 0x55
+      on_packet:
+        then:
+          - lambda: |-
+              ESP_LOGD("lambda", "packet %s", format_hex(x).c_str());
+
+      button:
+        - platform: template
+          name: "Transmit Packet"
+          on_press:
+            then:
+              - sx127x.send_packet:
+                 data: [0xC5, 0x51, 0x78, 0x82, 0xB7, 0xF9, 0x9C, 0x5C]
+
+FSK/OOK Continuous Mode:
+------------------------
+
+**As a Receiver:**
+
+The ``sx127x`` will output demodulated data onto DIO2,
+:doc:`remote_receiver </components/remote_receiver>` is used.
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    sx127x:
+      cs_pin: GPIO18
+      rst_pin: GPIO23
+      frequency: 433920000
+      modulation: OOK
+      rx_start: true
+      rx_bandwidth: 50_0kHz
+      rx_floor: -90
+
+    remote_receiver:
+      pin: GPIO32
+      dump: raw
+
+.. note::
+
+    It is not recommended to receive FSK in continuous mode as there will be too
+    much noise. Use packet mode instead.
+
+**As a Transmitter:**
+
+The ``sx127x`` expects raw data to be sent on DIO2,
```

Comment:
Would be great if this part could be explained a bit more or linked to some other documentation. 
Found some hints at https://github.com/LilyGO/ESP32-Paxcounter/blob/dee734a4893d4c4927dd8887d74be88b843d7df4/src/hal/ttgov2.h but before that it seemed almost as magic how data between sx127x and remote transmitter/receiver could just jump across without the sx127x defining use of that GPIO32